File: solutions/python/beer-song/1/beer_song.py
def recite(start, take=1):
    numBottles=start
    song = []
    while (take > 0):
        if numBottles > 2:
            line1 = str(numBottles) + " bottles of beer on the wall, " + \
                   str(numBottles) + " bottles of beer."
            line2 = "Take one down and pass it around, " + \
                   str(numBottles - 1) + " bottles of beer on the wall."
        elif numBottles == 2:
            line1 = str(numBottles) + " bottles of beer on the wall, " + \
                   str(numBottles) + " bottles of beer."
            line2 = "Take one down and pass it around, " + \
                     "1 bottle of beer on the wall."
        elif numBottles == 1:
            line1 = "1 bottle of beer on the wall, " + \
                   "1 bottle of beer."
            line2 = "Take it down and pass it around, " + \
                     "no more bottles of beer on the wall."
        else:
            line1 = "No more bottles of beer on the wall, " + \
                   "no more bottles of beer."
            line2 = "Go to the store and buy some more, " + \
                     "99 bottles of beer on the wall."
        song.append(line1)
        song.append(line2)

        if take > 1:
            song.append("")

        take-=1
        numBottles-=1
    return(song)

File: solutions/csharp/beer-song/1/BeerSong.cs
using System;

public static class BeerSong
{
    public static string Recite(int startBottles, int takeDown)
    {
        string song ="";
        for (int bb = startBottles; bb > startBottles - takeDown; bb--) {
            if (bb>0)
            {

                if (bb == 1)
                {
                    song += $"1 bottle of beer on the wall, 1 bottle of beer.\n";
                    song += "Take it down and pass it around, no more bottles of beer on the wall.";
                } else if (bb == 2)
                {
                    song += $"{bb} bottles of beer on the wall, {bb} bottles of beer.\n";
                    song += $"Take one down and pass it around, 1 bottle of beer on the wall.";
                } else
                {
                    song += $"{bb} bottles of beer on the wall, {bb} bottles of beer.\n";
                    song += $"Take one down and pass it around, {bb-1} bottles of beer on the wall.";
                }
            } else {
                song += "No more bottles of beer on the wall, no more bottles of beer.\n" +
                 "Go to the store and buy some more, 99 bottles of beer on the wall.";
            }
            song += "\n\n";
        }
        return song.Substring(0,song.Length-2);
    }
}